rust/src/pop3/pop3.rs

```diff
@@ -0,0 +1,498 @@
+/* Copyright (C) 2025 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use crate::applayer::*;
+use crate::conf::{conf_get, get_memval};
+use crate::core::{ALPROTO_FAILED, ALPROTO_UNKNOWN, IPPROTO_TCP};
+use crate::flow::Flow;
+use std;
+use std::collections::VecDeque;
+use std::ffi::CString;
+use std::os::raw::{c_char, c_int, c_void};
+use suricata_sys::sys::AppProto;
+
+use sawp::error::Error as SawpError;
+use sawp::error::ErrorKind as SawpErrorKind;
+use sawp::parser::Direction;
+use sawp::parser::Parse;
+use sawp::probe::Probe;
+use sawp::probe::Status;
+use sawp_pop3::{self, Command, ErrorFlag, Flag, Flags, InnerMessage, Response};
+
+static mut POP3_MAX_TX: usize = 256;
+
+pub(super) static mut ALPROTO_POP3: AppProto = ALPROTO_UNKNOWN;
+const POP3_PARSER: sawp_pop3::POP3 = sawp_pop3::POP3 {};
+
+#[derive(AppLayerEvent)]
+enum POP3Event {
+    TooManyTransactions,
```

Comment:
we also need a `pop3-events.rules` in our `rules/` dir

Reply:
Is there a standard wait to allocate the SID range for a new parser?

Reply:
Are events only for when parsing can't continue? Or would it make sense to add events for the error_flags as well?
eg: UnknownKeyword

Reply:
> Are events only for when parsing can't continue?

Not only

> Or would it make sense to add events for the error_flags as well? eg: UnknownKeyword

It can make sense

Reply:
If I convert ErrorFlags to Events, does it make sense to keep or remove the error fields?